### PR TITLE
Port examples to the new animation API

### PR DESCRIPTION
**Pull request by abarth**
These now use Animation and AnimationController instead of PerformanceView and
Performance.

### CODE REVIEW

File: examples/material_gallery/lib/demo/page_selector_demo.dart [hunk 1]
```diff
@@ -12,21 +12,21 @@ final List<String> _iconNames = <String>["event", "home", "android", "alarm", "f
 class TabViewDemo extends StatelessComponent {
   Widget _buildTabIndicator(BuildContext context, String iconName) {
     final Color color = Theme.of(context).primaryColor;
-    final AnimatedColorValue _selectedColor = new AnimatedColorValue(Colors.transparent, end: color, curve: Curves.ease);
-    final AnimatedColorValue _previousColor = new AnimatedColorValue(color, end: Colors.transparent, curve: Curves.ease);
+    final ColorTween _selectedColor = new ColorTween(begin: Colors.transparent, end: color);
```
Comment (Hixie) [inline examples/material_gallery/lib/demo/page_selector_demo.dart]: maybe unnamify the arguments

File: examples/widgets/smooth_resize.dart [hunk 2]
```diff
@@ -26,29 +26,28 @@ class SmoothBlock extends StatefulComponent {
...
       builder: (BuildContext context) {
+        double currentScale = scale.evaluate(animation);
         Matrix4 transform = new Matrix4.identity()
-          ..translate(x.value)
-          ..scale(scale.value, scale.value);
+          ..translate(x.evaluate(animation))
+          ..scale(currentScale, currentScale);
         return new Opacity(
-          opacity: opacity.value,
+          opacity: opacity.evaluate(animation),
```
Comment (Hixie) [inline examples/widgets/smooth_resize.dart]: Another option would be to convert the incoming arguments in the constructor into Evaluatable<double>s.

File: examples/widgets/smooth_resize.dart [hunk 3]
```diff
@@ -63,22 +62,22 @@ class SmoothBlockState extends State<SmoothBlock> {
...
+  Widget _handleEnter(Animation animation, Widget child) {
     return new CardTransition(
-      x: new AnimatedValue<double>(-200.0, end: 0.0, curve: Curves.ease),
-      opacity: new AnimatedValue<double>(0.0, end: 1.0, curve: Curves.ease),
-      scale: new AnimatedValue<double>(0.8, end: 1.0, curve: Curves.ease),
-      performance: performance,
+      x: new Tween<double>(begin: -200.0, end: 0.0),
+      opacity: new Tween<double>(begin: 0.0, end: 1.0),
+      scale: new Tween<double>(begin: 0.8, end: 1.0),
+      animation: new CurvedAnimation(parent: animation, curve: Curves.ease),
```
Comment (Hixie) [inline examples/widgets/smooth_resize.dart]: we could similarly precreate the `CurvedAnimation` in the constructor

Comment (abarth) [inline examples/widgets/smooth_resize.dart]: The `animation` comes in the callback parameter.

File: packages/flutter/lib/src/material/tabs.dart [hunk 4]
```diff
@@ -612,23 +606,27 @@ class _TabBarState<T> extends ScrollableState<TabBar<T>> implements TabBarSelect
...
       _valueIsChanging = true;
     }
-    Rect oldRect = _indicatorRect.value;
-    _indicatorRect.setProgress(_selection.performance.progress, AnimationDirection.forward);
-    Rect newRect = _indicatorRect.value;
-    if (oldRect != newRect)
+    Rect oldRect = _indicatorRect;
+    double t = _selection.animation.progress;
+    if (_valueIsChanging)
+      t = Curves.ease.transform(t);
```
Comment (Hixie) [inline packages/flutter/lib/src/material/tabs.dart]: add a comment explaining why we do this manually

**Comment by Hixie** [comment]
lgtm

File: examples/widgets/smooth_resize.dart (hunk 2 above)
Comment (abarth) [inline examples/widgets/smooth_resize.dart]: Done.

**Pull request merged**